op-node/rollup/derive/system_config.go

```diff
@@ -69,50 +81,86 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 	}
 	// indexed 1
 	updateType := ev.Topics[2]
-	// unindexed data
+
+	// Create a reader of the unindexed data
+	reader := bytes.NewReader(ev.Data)
+
+	// Helper function to prevent code duplication.
+	readWord := func() (b [32]byte) {
+		if _, err := reader.Read(b[:]); err != nil {
+			// The possible error returned by `Read` is ignored due to the length check of the unindexed data in
+			// all cases of the below switch statement. While we don't panic here, this log should *never* be emitted.
+			logger.Crit("failed to read word from unindexed log data")
+		}
+		return b
+	}
+
+	// Attempt to read unindexed data
 	switch updateType {
 	case SystemConfigUpdateBatcher:
 		if len(ev.Data) != 32*3 {
 			return fmt.Errorf("expected 32*3 bytes in batcher hash update, but got %d bytes", len(ev.Data))
 		}
-		if x := common.BytesToHash(ev.Data[:32]); x != (common.Hash{31: 32}) {
-			return fmt.Errorf("expected offset to point to length location, but got %s", x)
+
+		// Read the pointer, it should always equal 32.
+		if word := readWord(); common.BytesToHash(word[:]) != OneWordUint {
+			return fmt.Errorf("expected offset to point to length location, but got %s", word)
 		}
-		if x := common.BytesToHash(ev.Data[32:64]); x != (common.Hash{31: 32}) {
-			return fmt.Errorf("expected length of 1 bytes32, but got %s", x)
+
+		// Read the length, it should also always equal 32.
+		if word := readWord(); common.BytesToHash(word[:]) != OneWordUint {
+			return fmt.Errorf("expected length to be 32 bytes, but got %s", word)
 		}
-		if !bytes.Equal(ev.Data[64:64+12], make([]byte, 12)) {
-			return fmt.Errorf("expected version 0 batcher hash with zero padding, but got %x", ev.Data)
+
+		// Indexing `word` directly is always safe here, it is guaranteed to be 32 bytes in length.
+		// Check that the batcher address is correctly zero-padded.
+		word := readWord()
+		if !bytes.Equal(word[:12], AddressPadding) {
+			return fmt.Errorf("expected version 0 batcher hash with zero padding, but got %x", word)
 		}
-		destSysCfg.BatcherAddr.SetBytes(ev.Data[64+12:])
+		destSysCfg.BatcherAddr.SetBytes(word[12:])
 		return nil
-	case SystemConfigUpdateGasConfig: // left padded uint8
+	case SystemConfigUpdateGasConfig:
 		if len(ev.Data) != 32*4 {
 			return fmt.Errorf("expected 32*4 bytes in GPO params update data, but got %d", len(ev.Data))
 		}
-		if x := common.BytesToHash(ev.Data[:32]); x != (common.Hash{31: 32}) {
-			return fmt.Errorf("expected offset to point to length location, but got %s", x)
+
+		// Read the pointer, it should always equal 32.
+		if word := readWord(); common.BytesToHash(word[:]) != OneWordUint {
+			return fmt.Errorf("expected offset to point to length location, but got %s", word)
 		}
-		if x := common.BytesToHash(ev.Data[32:64]); x != (common.Hash{31: 64}) {
-			return fmt.Errorf("expected length of 2 bytes32, but got %s", x)
+
+		// Read the length, it should always equal 64.
+		if word := readWord(); common.BytesToHash(word[:]) != OneWordUint {
```

Comment:
```suggestion
		if word := readWord(); word != OneWordUint {
```

Copy-paste bug of `OneWordUint`? It was checking a 64 value before. cc @tynes are you really sure this is more readable? I liked the previous value checks, it's shorter too.

Reply:
Ah yes- good spot